`tailscale netcheck`: nil pointer when there are no DERP responses

### What is the issue?

C:\Users\aleja>tailscale netcheck

Report:
        * UDP: true
        * IPv4: yes, 45.238.177.116:54882
        * IPv6: no, but OS has support
        * MappingVariesByDestIP: false
        * HairPinning: false
        * PortMapping:
panic: runtime error: invalid memory address or nil pointer dereference
[signal 0xc0000005 code=0x0 addr=0x18 pc=0x102c25f]

goroutine 1 [running]:
tailscale.com/cmd/tailscale/cli.printReport(0xc0001da0d0, 0xc000336160)
        tailscale.com@v1.40.0/cmd/tailscale/cli/netcheck.go:151 +0x53f
tailscale.com/cmd/tailscale/cli.runNetcheck({0x1251828, 0xc0000240d8}, {0x0?, 0x0?, 0xc00006bf40?})
        tailscale.com@v1.40.0/cmd/tailscale/cli/netcheck.go:91 +0x494
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0x0?, {0x1251828?, 0xc0000240d8?})
        github.com/peterbourgon/ff/v3@v3.1.2/ffcli/command.go:153 +0x16c
github.com/peterbourgon/ff/v3/ffcli.(*Command).Run(0xc000144630?, {0x1251828?, 0xc0000240d8?})
        github.com/peterbourgon/ff/v3@v3.1.2/ffcli/command.go:157 +0x113
tailscale.com/cmd/tailscale/cli.Run({0xc00006c3d0?, 0xc00007c000?, 0xc00005e000?})
        tailscale.com@v1.40.0/cmd/tailscale/cli/cli.go:173 +0x905
main.main()
        tailscale.com@v1.40.0/cmd/tailscale/tailscale.go:22 +0xfb


### Steps to reproduce

_No response_

### Are there any recent changes that introduced the issue?

_No response_

### OS

Windows

### OS version

Windows 11

### Tailscale version

1.40.0   tailscale commit: 105be684dd04727a58ed213ec9499d071a377ece   other commit: 4c70014a47f1816e1573a8cef58cdaf96f78a264   go version: go1.20.3-tsddff070

### Other software

_No response_

### Bug report

_No response_

--- a/cmd/tailscale/cli/netcheck.go
+++ b/cmd/tailscale/cli/netcheck.go
@@ -148,7 +148,11 @@ func printReport(dm *tailcfg.DERPMap, report *netcheck.Report) error {
 	if len(report.RegionLatency) == 0 {
 		printf("\t* Nearest DERP: unknown (no response to latency probes)\n")
 	} else {
-		printf("\t* Nearest DERP: %v\n", dm.Regions[report.PreferredDERP].RegionName)
+		if report.PreferredDERP != 0 {
+			printf("\t* Nearest DERP: %v\n", dm.Regions[report.PreferredDERP].RegionName)
+		} else {
+			printf("\t* Nearest DERP: [none]\n")
+		}
 		printf("\t* DERP latency:\n")
 		var rids []int
 		for rid := range dm.Regions {

